lib/xml/encrypt.js

```diff
@@ -39,13 +39,13 @@ exports.encrypted = function (encryptOptions) {
                 pem: utils.fixPemFormatting(options.encryptionCert)
             },
             function (retryErr, retryEncrypted) {
-          err = retryErr;
-          encrypted = retryEncrypted;
+              if (retryErr) return callback(retryErr)
+              return callback(null, utils.removeWhitespace(retryEncrypted));
         })
```

Comment:
you'll want to add a `return` after this call to `xmlenc.encrypt` to avoid the callback being called on both lines 43 and 48. 

I also believe that it should be possible to remove line 47 since this new branch will handle the error case.

Reply:
I re-worked it a bit because it got messy fast trying to deal with the callbacks.